controllers/mentor_test.go

```diff
@@ -196,3 +197,141 @@ func TestFetchMentorOK(t *testing.T) {
 		}
 	}
 }
+
+// Test unauthenticated request to /mentor/dashboard/
+func TestMentorDashboardNoAuth(t *testing.T) {
+	testRequestNoAuth(t, "GET", "/mentor/dashboard/")
+}
+
+// Test request to /mentor/dashboard/ with invalid jwt
+func TestMentorDashboardInvalidAuth(t *testing.T) {
+	testRequestInvalidAuth(t, "GET", "/mentor/dashboard/")
+}
+
+// Test unauthenticated request to /mentor/dashboard/ with no registration
+func TestMentorDashboardNoReg(t *testing.T) {
+	// Set up a local test database path
+	db := setTestDB()
+	defer unsetTestDB()
+
+	// Generate a jwt secret key for testing
+	setTestJwtSecretKey()
+	defer unsetTestJwtSecretKey()
+
+	// Test login fields
+	testUsername := getTestUsername()
+	testLoginFields := utils.LoginJwtFields{Username: testUsername}
+
+	testJwt, _ := utils.GenerateLoginJwtString(testLoginFields)
+
+	req, _ := http.NewRequest(
+		"GET",
+		"/mentor/dashboard/",
+		nil,
+	)
+	req.Header.Add("Bearer", testJwt)
+
+	res := executeRequest(req, db)
+
+	expectStatusCodeToBe(t, res, http.StatusBadRequest)
+	expectResponseBodyToBe(t, res, fmt.Sprintf("Mentor `%s` does not exists.", testUsername))
+}
+
+// Test requests to /mentor/dashboard/ with registered and  proper authentication
+func TestMentorDashboardOK(t *testing.T) {
+	// Set up a local test database path
+	db := setTestDB()
+	defer unsetTestDB()
+
+	// Generate a jwt secret key for testing
+	setTestJwtSecretKey()
+	defer unsetTestJwtSecretKey()
+
+	// Test login fields
+	testUsername := getTestUsername()
+	testLoginFields := utils.LoginJwtFields{Username: testUsername}
+
+	testJwt, _ := utils.GenerateLoginJwtString(testLoginFields)
+
+	modelMentor := models.Mentor{
+		Name:     "TestMentor",
+		Email:    "iamamentor@cool.com",
+		Username: testUsername,
+	}
+
+	db.Table("mentors").Create(&modelMentor)
+
+	testProjects := generateTestProjects(5, false, true)
+	testProjects[1].MentorId = int32(modelMentor.ID)
+	testProjects[3].SecondaryMentorId = int32(modelMentor.ID)
+
+	var projects []controllers.ProjectInfo
+	var students []controllers.StudentInfo
+
+	modelStudents := generateTestStudents(5)
+
+	for i, student := range modelStudents {
+		if i < 3 {
+			testProjects[1].Contributors = testProjects[1].Contributors + student.Username + ","
+		} else {
+			testProjects[3].Contributors = testProjects[3].Contributors + student.Username + ","
```

Comment:
that will result in a trailing comma and `strings.Split` will later make an extra empty string at the end of this slice